File: frontend/src/api/dict.ts
import request from './request';

export interface DictType {
  id: string;
  code: string;
  name: string;
  description: string;
  sort_order: number;
  status: 0 | 1;
  is_system: boolean;
  created_at: string;
  updated_at: string;
}

export interface DictItem {
  id: string;
  type_id: string;
  type_code: string;
  item_value: string;
  item_label: string;
  sort_order: number;
  status: 0 | 1;
  created_at: string;
  updated_at: string;
}

export interface CreateDictTypeParams {
  code: string;
  name: string;
  description?: string;
  sort_order?: number;
  status?: 0 | 1;
}

export interface UpdateDictTypeParams {
  name?: string;
  description?: string;
  sort_order?: number;
  status?: 0 | 1;
}

export interface CreateDictItemParams {
  type_code: string;
  item_value: string;
  item_label: string;
  sort_order?: number;
  status?: 0 | 1;
}

export interface UpdateDictItemParams {
  item_label?: string;
  sort_order?: number;
  status?: 0 | 1;
}

export function getDictTypes(): Promise<DictType[]> {
  return request.get('/system/dicts/types');
}

export function createDictType(data: CreateDictTypeParams): Promise<DictType> {
  return request.post('/system/dicts/types', data);
}

export function updateDictType(id: string, data: UpdateDictTypeParams): Promise<DictType> {
  return request.put(`/system/dicts/types/${id}`, data);
}

export function deleteDictType(id: string): Promise<void> {
  return request.delete(`/system/dicts/types/${id}`);
}

export function getDictItems(typeCode: string, all = false): Promise<DictItem[]> {
  return request.get(`/system/dicts/${typeCode}`, { params: all ? { all: 1 } : undefined });
}

export function createDictItem(data: CreateDictItemParams): Promise<DictItem> {
  return request.post('/system/dicts', data);
}

export function updateDictItem(id: string, data: UpdateDictItemParams): Promise<DictItem> {
  return request.put(`/system/dicts/${id}`, data);
}

export function deleteDictItem(id: string): Promise<void> {
  return request.delete(`/system/dicts/${id}`);
}

File: backend/internal/dict/dto/dict.go
package dto

import "time"

type TypeResponse struct {
	ID          string    `json:"id"`
	Code        string    `json:"code"`
	Name        string    `json:"name"`
	Description string    `json:"description"`
	SortOrder   int       `json:"sort_order"`
	Status      int16     `json:"status"`
	IsSystem    bool      `json:"is_system"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type ItemResponse struct {
	ID        string    `json:"id"`
	TypeID    string    `json:"type_id"`
	TypeCode  string    `json:"type_code"`
	ItemValue string    `json:"item_value"`
	ItemLabel string    `json:"item_label"`
	SortOrder int       `json:"sort_order"`
	Status    int16     `json:"status"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

type CreateTypeRequest struct {
	Code        string `json:"code" binding:"required,max=50"`
	Name        string `json:"name" binding:"required,max=100"`
	Description string `json:"description" binding:"max=255"`
	SortOrder   int    `json:"sort_order"`
	Status      int16  `json:"status" binding:"omitempty,oneof=0 1"`
}

type UpdateTypeRequest struct {
	Name        *string `json:"name" binding:"omitempty,max=100"`
	Description *string `json:"description" binding:"omitempty,max=255"`
	SortOrder   *int    `json:"sort_order"`
	Status      *int16  `json:"status" binding:"omitempty,oneof=0 1"`
}

type CreateItemRequest struct {
	TypeCode  string `json:"type_code" binding:"required,max=50"`
	ItemValue string `json:"item_value" binding:"required,max=50"`
	ItemLabel string `json:"item_label" binding:"required,max=100"`
	SortOrder int    `json:"sort_order"`
	Status    int16  `json:"status" binding:"omitempty,oneof=0 1"`
}

type UpdateItemRequest struct {
	ItemLabel *string `json:"item_label" binding:"omitempty,max=100"`
	SortOrder *int    `json:"sort_order"`
	Status    *int16  `json:"status" binding:"omitempty,oneof=0 1"`
}
